README.md

```diff
@@ -1,4 +1,241 @@
-# orm-mapping-to-tabel
+# ORM: Mapping Ruby Classes to Database Tables
+
+## Objectives
+
+1. Understand the concept of mapping a Ruby class to a database table and an instance of a class to a table row.
+2. Learn how to write code that maps a Ruby class to a database table. 
+3. Learn how to write code that inserts data regarding an instance of a class into a database table row. 
+
+## Mapping a Class to a Table
+
+When building an ORM to connect our Ruby program to a database, we equate a class with a database table and the instances that the class produces to rows in that table. 
+
+Let's say we are building a music player app that allows users to store their music and browse their songs by song.
+
+This program will have a `Song` class. Each song instance will have a name and an album attribute. 
+
+```ruby
+class Song
+
+  attr_accessor :name, :album
+  
+  def initialize(name, album)
+    @name = name
+    @album = album
+  end
+
+end
+```
+
+Here we have an `attr_accessor` for `name` and `album`. In order to "map" this `Song` class to a songs database table, we need to create our database, then we need to create our songs table. 
+
+### Creating the Database
+
+Before we can create a songs table we need to create our music database. Whose responsibility is it create the database? It is not the responsibility of our `Song` class. Remember, classes are mapped to *tables inside a database*, not to the database as a whole. We may want to build other classes that we equate with other database tables later on. 
+
+It is the responsibility of our program as a whole to create and establish the database. Accordingly, you'll see our Ruby programs set up such that they have a `config` directory that contains an `environment.rb` file. This file will look something like this:
+
+```ruby
+require 'sqlite3'
+require_relative '../lib/song.rb'
+
+DB = {:conn => SQLite3::Database.new("db/music.db")}
+```
+
+Here we set up a constant, `DB`, that is equal to a hash that contains our connection to the database. In our `lib/song.rb` file, we can therefore access the `DB` constant and the database connection it holds like this:
+
+```ruby
+DB[:conn]
+```
+
+So, as we move through this reading, let's assume that our hypothetical program has just such a `config/environment.rb` file and that the `DB[:conn]` constant refers to our connection to the database. 
+
+Now that our hypothetical database is set up in our hypothetical program, let's move on to our `Song` class and it's equivalent database table. 
+
+### Creating the Table
+
+According to the ORM convention in which a class is mapped to or equated with a database table, we need to create a songs table. We will accomplish this by writing a class method in our `Song` class that creates this table. 
```

Comment:
Mention that the ORM convention is to pluralize the class name into a table name